adb/adb_commands.py

```diff
@@ -167,25 +234,22 @@ def Push(self, source_file, device_filename, mtime='0', timeout_ms=None, progres
       progress_callback: callback method that accepts filename, bytes_written and total_bytes,
                          total_bytes will be -1 for file-like objects
     """
-    should_close = False
     if isinstance(source_file, str):
       if os.path.isdir(source_file):
         self.Shell("mkdir " + device_filename)
         for f in os.listdir(source_file):
           self.Push(os.path.join(source_file, f), device_filename + '/' + f, progress_callback=progress_callback)
         return
       source_file = open(source_file, "rb")
-      should_close = True
 
-    connection = self.protocol_handler.Open(
+    with source_file:
+      connection = self.protocol_handler.Open(
         self.handle, destination=b'sync:', timeout_ms=timeout_ms)
-    self.filesync_handler.Push(connection, source_file, device_filename,
+      self.filesync_handler.Push(connection, source_file, device_filename,
                                mtime=int(mtime), progress_callback=progress_callback)
-    if should_close:
-      source_file.close()
     connection.Close()
 
-  def Pull(self, device_filename, dest_file='', progress_callback=None, timeout_ms=None):
+  def Pull(self, device_filename, dest_file=None, timeout_ms=None, progress_callback=None):
```

Comment:
why reorder the args?

Reply:
progress_callback was a very recent parameter addition that bumped the order of timeout_ms, which was affecting callers that used previous versions of this and expected timeout_ms to be the 3rd argument

Reply:
Previous versions of this library had timeout_ms as the 3rd argument. The recent addition of progress_callback broke the order in which the arguments were expected, so I push the new argument back